backend/de.metas.business/src/main/java/de/metas/costing/ICostDetailService.java

```diff
@@ -38,7 +38,7 @@
 
 	boolean hasCostDetailsForProductId(ProductId productId);
 
-	Optional<CostDetail> getExistingCostDetail(CostDetailCreateRequest request);
+	Optional<List<CostDetail>> getExistingCostDetails(CostDetailCreateRequest request);
```

Comment:
why not using `List<CostDetail>`?

what's the difference between Optional.emty() and Optional of empty list?